gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/ReminderSettingDagProcLeaseArbiter.java

```diff
@@ -0,0 +1,172 @@
+   * lease for a particular job event and return the status of the attempt.
+   * @param flowAction
+   * @param eventTimeMillis
+   * @param isReminderEvent
+   * @param skipFlowExecutionIdReplacement
+   * @return
+   */
+  @Override
+  public MultiActiveLeaseArbiter.LeaseAttemptStatus tryAcquireLease(DagActionStore.DagAction flowAction, long eventTimeMillis,
+      boolean isReminderEvent, boolean skipFlowExecutionIdReplacement) {
+    try {
+      MultiActiveLeaseArbiter.LeaseAttemptStatus leaseAttemptStatus = this.tryAcquireLease(flowAction, eventTimeMillis, isReminderEvent, skipFlowExecutionIdReplacement);
+      /* Schedule a reminder for the event unless the lease has been completed to safeguard against case lease owner
+      fails to complete lease
+      */
+      if (leaseAttemptStatus instanceof MultiActiveLeaseArbiter.LeaseObtainedStatus) {
+        scheduleReminderForEvent((LeaseObtainedStatus) leaseAttemptStatus);
+      } else if (leaseAttemptStatus instanceof  MultiActiveLeaseArbiter.LeasedToAnotherStatus) {
+        scheduleReminderForEvent((LeasedToAnotherStatus) leaseAttemptStatus);
+      }
+      return leaseAttemptStatus;
+    } catch (SchedulerException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public boolean recordLeaseSuccess(LeaseObtainedStatus status)
+      throws IOException {
+    return this.decoratedLeaseArbiter.recordLeaseSuccess(status);
+  }
+
+  protected void scheduleReminderForEvent(MultiActiveLeaseArbiter.LeasedToAnotherStatus leaseStatus)
+      throws SchedulerException {
+    _dagActionReminderScheduler.scheduleReminder(leaseStatus.getDagAction(), leaseStatus.getMinimumLingerDurationMillis());
+  }
+
+  protected void scheduleReminderForEvent(MultiActiveLeaseArbiter.LeaseObtainedStatus leaseStatus)
+      throws SchedulerException {
+    _dagActionReminderScheduler.scheduleReminder(leaseStatus.getDagAction(), leaseStatus.getMinimumLingerDurationMillis());
+  }
+
+  @Slf4j
+  public static class ReminderJob implements Job {
+    public static final String FLOW_ACTION_TYPE_KEY = "flow.actionType";
+    public static final String DAG_MANAGEMENT_KEY = "dag.management";
+
+    @Override
+    public void execute(JobExecutionContext context) {
+      // Get properties from the trigger to create a dagAction
+      JobDataMap jobDataMap = context.getTrigger().getJobDataMap();
+      String flowName = jobDataMap.getString(ConfigurationKeys.FLOW_NAME_KEY);
+      String flowGroup = jobDataMap.getString(ConfigurationKeys.FLOW_GROUP_KEY);
+      String jobName = jobDataMap.getString(ConfigurationKeys.JOB_NAME_KEY);
+      String flowId = jobDataMap.getString(ConfigurationKeys.FLOW_EXECUTION_ID_KEY);
+      DagActionStore.DagActionType dagActionType = DagActionStore.DagActionType.valueOf(
+          jobDataMap.getString(FLOW_ACTION_TYPE_KEY));
+      DagManagement dagManagement = (DagManagement) jobDataMap.get(DAG_MANAGEMENT_KEY);
```

Comment:
I really would like to avoid this (de)serialization.  perhaps we should explore a `@Singleton` with an accessor for the "global" `DagManagement`

Reply:
It's already a singleton class but the issue lies in `ReminderJob` class & methods being static so there's no reference to `DagManagement ` passed directly